examples/multigpu_advtrain/target_attack.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np
+import tensorflow as tf
+
+from cleverhans.model import Model, CallableModelWrapper
+from cleverhans.attacks import Attack, FastGradientMethod
+from cleverhans.utils_tf import model_loss
+
+from model import clone_variable, MLP_probs
+
+
+class ProjectedGradientDescentMethod(Attack):
+
+    """
+    The Basic Iterative Method (Kurakin et al. 2016). The original paper used
+    hard labels for this attack; no label smoothing.
+    Paper link: https://arxiv.org/pdf/1607.02533.pdf
+    """
+
+    def __init__(self, model, back='tf', sess=None):
+        """
+        Create a ProjectedGradientDescentMethod instance.
+        """
+        super(ProjectedGradientDescentMethod, self).__init__(model, back, sess)
+        self.feedable_kwargs = {'eps': np.float32,
+                                'eps_iter': np.float32,
+                                'y': np.float32,
+                                'y_target': np.float32,
+                                'clip_min': np.float32,
+                                'clip_max': np.float32}
+        self.structural_kwargs = ['ord', 'nb_iter']
+
+        if not isinstance(self.model, Model):
+            self.model = MLP_probs(self.model)
+
+    def clip_eta(self, eta):
```

Comment:
This is now a standalone function in utils_tf.py

Reply:
Fixed in 460d3f3.